src/dappsExplorer/utils.tsx

```diff
@@ -0,0 +1,26 @@
+import { DappV1, DappV2, isDappV2 } from 'src/dapps/types'
+
+/***
+ * Calculates a score based on the search term and dapp
+ * - +2 if the name matches the search term in a case sensitive way
+ * - +1 if the name matches the search term in a case insensitive way
+ * - +1 if the description matches the search term in a case insensitive way
+ * - +1 if the category matches the search term in a case insensitive way
+ * @param dapp
+ * @param searchTerm
+ * @returns {number} the score of the dapp 0 - 5
+ */
+export const calculateSearchScore = (dapp: DappV2 | DappV1, searchTerm: string) => {
+  if (searchTerm === '') return 0
+  const nameMatchingCaseScore = dapp.name.includes(searchTerm) ? 2 : 0
+  const nameScore = dapp.name.toLowerCase().includes(searchTerm.toLowerCase()) ? 1 : 0
+  const descriptionScore = dapp.description.toLowerCase().includes(searchTerm.toLowerCase()) ? 1 : 0
+  const categoryScore = isDappV2(dapp)
+    ? dapp.categories.some((category) => category.toLowerCase().includes(searchTerm.toLowerCase()))
+      ? 1
+      : 0
+    : dapp.categoryId.toLowerCase().includes(searchTerm.toLowerCase())
+    ? 1
+    : 0
+  return nameMatchingCaseScore + nameScore + descriptionScore + categoryScore
```

Comment:
do we want to include the category in the scoring? i wonder if this would be confusing - for example, the user has no filters selected and they search for "swap". in this case, bitssa would be displayed but it's not clear why since there's no "swap" in the name or description :/ i wonder if this would look like a broken search, since in this scenario we do not display the dapp category anywhere.

i figure that you probably included this so that people can search for categories and get a list of matching dapps, but this is what the filters are for right? if we want the search to do something similar, it kind of feels like we should surface the category of the dapp on the dapp card itself to make it less confusing why the search result includes the dapp? 

idk, just my 2 cents. maybe Reed has an opinion on this too

Reply:
Yeah I agree searching on the category id is is not optimal. I'd expect as a user if I type swap to see bittsa since they provide swaps. What would be better is category name because then I would be localized.

Reply:
@kathaypacific added the localized category name to search in e2fe4326379d19296c03a40a63c76f33bd2df86a.

We could remove considering the category name entirely, but this might lead to some strange interactions between search and filter & search.

#### Without Fix or without Categories considered Search Example:
- User Searches 'Usa cripto'
- Dapps Listed: None

#### With Fix or without Categories considered Search Example:
- User Searches 'Usa cripto'
- Dapps Listed: Bidali, Flywallet, ChiSpend

#### Without Fix of without Categories Filter & Search Example:
- User Searches 'Usa cripto'
- User selects 'Usa cripto' Filter
- Dapps Listed: None
-

Reply:
I agree with Kathy, I think the categories should be filtered only using the filter chips. Or, we have to add some indication in the UI so it's clear when there is a match because of the category.
But I don't know if you have already talked about this